### DISCUSSION TITLE
questions on cachenode

### DISCUSSION THREAD

**Discussion by zhouhanAC** [zeromicro/go-zero#1304] [Q&A]
在redis.go中 Del方法内部的getRedis已经区分了cachenode.redis是ClusterType和NodeType类型，这两种类型都实现了Cmdable的接口，这个接口中的Del方法可以删除单个key和批量删除keys。那为什么在cachenode.go:Del（line:63)这个函数里还要区分cachenode.redis的类型，并且在Type为redis.ClusterType要一个个删除key而不是批量删除key?没看明白，求大佬解答一下

**Comment by kevwan** [comment]
因为这些key不一定在同一个node上，一次删可能删不全

**Reply by zhouhanAC** [reply]
了解了，redis集群删除key如果在不同节点 上是需要登陆到对应节点 删除的。
另cachenode_test.go第二个测试用例有点问题：
  func TestCacheNode_DelCacheWithErrors(t *testing.T) {
	  store, clean, err := redistest.CreateRedis()
	  assert.Nil(t, err)
	  store.Type = redis.ClusterType
	  clean()

	cn := cacheNode{
		rds:            store,
		r:              rand.New(rand.NewSource(time.Now().UnixNano())),
		lock:           new(sync.Mutex),
		unstableExpiry: mathx.NewUnstable(expiryDeviation),
		stat:           NewStat("any"),
		errNotFound:    errTestNotFound,
	}
	assert.Nil(t, cn.Del("third", "fourth"))
}
过早的clean()关闭了redis。应该是defer clean()，大佬笔误了吧

**Reply by kevwan** [reply]
是笔误了，你可以PR一下哈，不过可以改成这样：

```go
store, clean, err := redistest.CreateRedis()
assert.Nil(t, err)
defer clean()
store.Type = redis.ClusterType
```
